presto-main/src/main/java/com/facebook/presto/operator/TopNOperator.java

```diff
@@ -106,17 +104,17 @@ public TopNOperator(
         this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
         this.localUserMemoryContext = operatorContext.localUserMemoryContext();
         checkArgument(n >= 0, "n must be positive");
-
         if (n == 0) {
             finishing = true;
-            outputIterator = emptyIterator();
+            outputPages = WorkProcessor.of();
         }
         else {
-            topNBuilder = new GroupedTopNBuilder(
+            topNBuilder = new InMemoryGroupedTopNBuilder(
                     types,
                     new SimplePageWithPositionComparator(types, sortChannels, sortOrders),
                     n,
                     false,
+                    localUserMemoryContext,
```

Comment:
Given we are passing in `localUserMemoryContext`, we do not need `localUserMemoryContext` for class `TopNOperator`.

Reply:
Sorry, could you rephrase that ? I do not understand the comment.

Reply:
`localUserMemoryContext` is a final private member of the class. Just remove it.

Reply:
Ah! My bad. I see what you mean 👍🏼